src/dotnet-interactive-vscode/insiders/src/notebookControllers.ts

```diff
@@ -146,8 +146,10 @@ export class DotNetNotebookKernel {
     }
 
     private async executeHandler(cells: vscode.NotebookCell[], document: vscode.NotebookDocument, controller: vscode.NotebookController): Promise<void> {
-        for (const cell of cells) {
-            await this.executeCell(cell, controller);
+        if (vscode.workspace.isTrusted) {
+            for (const cell of cells) {
+                await this.executeCell(cell, controller);
+            }
```

Comment:
Might be good to show an information dialog that execution won't happen because it's untrusted, otherwise a user won't have any idea why nothing happens.  Would be good to coordinate with Python on the experience.

Reply:
if you hit the run button in vscode you get the dialog you see in the screenshot above

Reply:
removed